### PR TITLE
WIP: Pull Out Conversion Factors and Renaming

### PR DESCRIPTION
**Pull request by BrettBoval**
Refactor Part of WebUIAdaptor for Edge Cases and Readability

### Please Check
- [ ] Are the [JSON schemas](https://github.com/covid-projections/covid-data-model/tree/master/api/schemas) updated?
- [ ] Are the [api docs](https://github.com/covid-projections/covid-data-model/blob/master/api/README.V1.md) updated?
- [ ] Are tests passing?

### CODE REVIEW

**Comment by BrettBoval** [comment]
@jeremiq Did you have better names for these variables? I think the descriptors are still confusing. Either have simpler variable names with more comments or think of better names.

File: pyseir/deployment/model_conversion_factors.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+from datetime import timedelta
+
+from pyseir import load_data
+
+log = structlog.get_logger()
+
+UNITY_SCALING_FACTOR = 1
+
+
+def _get_model_to_dataset_conversion_factors_for_state(state, t0_simulation, fips):
```
Comment (jeremiq) [inline pyseir/deployment/model_conversion_factors.py:12]: add some type annotations, state needs to be a string that's either a full name or a valid abbreviation.

File: pyseir/deployment/model_conversion_factors.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        t0=t0_simulation,
+        category=load_data.HospitalizationCategory.HOSPITALIZED,
+    )
+
+    if observed_latest_hospitalized is None:
+        # We have no observed data available. Best we can do is pass unity factors.
+        return UNITY_SCALING_FACTOR, UNITY_SCALING_FACTOR
+    elif observed_latest_hospitalized == 0:
+        # Right now our scaling factor can not capture this edge case
+        log.msg(
```
Comment (jeremiq) [inline pyseir/deployment/model_conversion_factors.py:44]: log with a level here, probably info

File: pyseir/deployment/model_conversion_factors.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+            "Observed Hospitalized was 0 so we can not scale model outputs to latest observed",
+            state=state,
+        )
+        return UNITY_SCALING_FACTOR, UNITY_SCALING_FACTOR
+    else:
+        # Let's try to get a conversion for model to observed hospitalization
+
+        # Rebuild date object
+        t_latest_hosp_data_date = t0_simulation + timedelta(days=int(days_since_start))
```
Comment (jeremiq) [inline pyseir/deployment/model_conversion_factors.py:53]: you've already got 'date' in the name, can we kill the `t_`?

File: pyseir/deployment/model_conversion_factors.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+    Parameters
+    ----------
+    state
+    t0_simulation
+    fips
+
+    Returns
+    -------
+    convert_model_to_observed_hospitalized
```
Comment (jeremiq) [inline pyseir/deployment/model_conversion_factors.py:25]: I actually like the names you have down below, ` model_to_observed_hospitalized_ratio, model_to_observed_icu_ratio`

File: pyseir/deployment/model_conversion_factors.py [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+            category=load_data.HospitalizationCategory.ICU,
+        )
+        if observed_latest_icu is None:
+            # We have observed hospitalizations, but not observed icu
+            # We therefore scale ICU the same as general hospitalization
+            model_to_observed_icu_ratio = model_to_observed_hospitalized_ratio
+            return model_to_observed_hospitalized_ratio, model_to_observed_icu_ratio
+        elif observed_latest_icu == 0:
+            # Right now our scaling factor can not capture this edge case
+            log.msg(
```
Comment (jeremiq) [inline pyseir/deployment/model_conversion_factors.py:82]: same here, log with a real level.

File: pyseir/deployment/model_conversion_factors.py [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+        )
+
+        # In the model, general hospital and icu hospital are disjoint states. We have to add them
+        # together to get the correct comparable for hospitalized.
+        model_heads_in_beds = model_state_hosp_gen + model_state_hosp_icu
+
+        model_to_observed_hospitalized_ratio = observed_latest_hospitalized / model_heads_in_beds
+
+        # Now let's look at ICU observed data
+        _, observed_latest_icu = load_data.get_current_hospitalized_for_state(
```
Comment (jeremiq) [inline pyseir/deployment/model_conversion_factors.py:70]: match the method name, call this `observed_current_icu` or change the method name ;)

File: pyseir/deployment/model_conversion_factors.py [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+        # Let's try to get a conversion for model to observed hospitalization
+
+        # Rebuild date object
+        t_latest_hosp_data_date = t0_simulation + timedelta(days=int(days_since_start))
+
+        # Get Compartment Values for a Given Time
+        model_state_hosp_gen = load_data.get_compartment_value_on_date(
+            fips=fips, compartment="HGen", date=t_latest_hosp_data_date
+        )
+        model_state_hosp_icu = load_data.get_compartment_value_on_date(
```
Comment (jeremiq) [inline pyseir/deployment/model_conversion_factors.py:59]: everything in this function only applies to the state, so get rid of _state_ in the name, for simplicity. Also write out `hosp` and `gen` for added clarity

File: pyseir/deployment/model_conversion_factors.py [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+    -------
+    convert_model_to_observed_hospitalized
+    convert_model_to_observed_icu
+    """
+
+    # Get "Ground Truth" from outside datasets
+    # NB: If only cumulatives are provided, we estimate current load. So this isn't strictly
+    # actuals from covid-tracking.
+    state_abbreviation = us.states.lookup(state).abbr
+    days_since_start, observed_latest_hospitalized = load_data.get_current_hospitalized_for_state(
```
Comment (jeremiq) [inline pyseir/deployment/model_conversion_factors.py:33]: match the method name latest --> current

File: pyseir/deployment/model_conversion_factors.py [hunk 9]
```diff
@@ -0,0 +1,102 @@
+import us
+import structlog
+from datetime import timedelta
+
+from pyseir import load_data
+
+log = structlog.get_logger()
+
+UNITY_SCALING_FACTOR = 1
```
Comment (jeremiq) [inline pyseir/deployment/model_conversion_factors.py:9]: you may be going overboard here ;), good!

**Pull request closed without merging**